grafana-plugin/pkg/plugin/resources.go

```diff
@@ -2,36 +2,231 @@ package plugin
+	ID          int    `json:"UserID"`
+	IsAnonymous bool   `json:"IsAnonymous"`
+	Name        string `json:"Name"`
+	Login       string `json:"Login"`
+	Email       string `json:"Email"`
+	Role        string `json:"Role"`
+}
+
+type UserIDJSONData struct {
+	ID int `json:"intField"`
+}
+
+func SetXInstanceContextHeader(settings OnCallPluginSettings, req *http.Request) error {
+	xInstanceContext := XInstanceContextJSONData{
+		StackId:      strconv.Itoa(settings.StackID),
+		OrgId:        strconv.Itoa(settings.OrgID),
+		GrafanaToken: settings.GrafanaToken,
+	}
+	xInstanceContextHeader, err := json.Marshal(xInstanceContext)
+	if err != nil {
+		return err
+	}
+	req.Header.Set("X-Instance-Context", string(xInstanceContextHeader))
+	return nil
+}
+
+func SetXGrafanaContextHeader(user *backend.User, userID int, req *http.Request) error {
+	var xGrafanaContext XGrafanaContextJSONData
+	if user == nil {
+		xGrafanaContext = XGrafanaContextJSONData{
+			IsAnonymous: true,
+		}
+	} else {
+		xGrafanaContext = XGrafanaContextJSONData{
+			ID:          userID,
+			IsAnonymous: false,
+			Name:        user.Name,
+			Login:       user.Login,
+			Email:       user.Email,
+			Role:        user.Role,
+		}
+	}
+	xGrafanaContextHeader, err := json.Marshal(xGrafanaContext)
+	if err != nil {
+		return err
+	}
+	req.Header.Set("X-Grafana-Context", string(xGrafanaContextHeader))
+	return nil
+}
+
+func SetAuthorizationHeader(settings OnCallPluginSettings, req *http.Request) {
+	req.Header.Set("Authorization", settings.OnCallToken)
+}
+
+func (a *App) GetUserID(user *backend.User, settings OnCallPluginSettings) (int, error) {
+	reqURL, err := url.Parse(settings.GrafanaURL)
+	if err != nil {
+		return 0, err
+	}
+
+	reqURL.Path += "api/users"
+	q := reqURL.Query()
+	q.Set("login", user.Login)
+	reqURL.RawQuery = q.Encode()
+
+	req, err := http.NewRequest("GET", reqURL.String(), nil)
+	if err != nil {
+		return 0, err
+	}
+
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", settings.GrafanaToken))
+
+	res, err := a.httpClient.Do(req)
+	if err != nil {
+		return 0, err
+	}
+	defer res.Body.Close()
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return 0, err
+	}
+
+	log.DefaultLogger.Info(fmt.Sprintf("User Response %s %s %s", reqURL.String(), res.Status, body))
+
+	var result []map[string]interface{}
+	err = json.Unmarshal(body, &result)
+	if err != nil {
+		err = fmt.Errorf("Error unmarshalling JSON: %+v", err)
+		log.DefaultLogger.Error(err.Error())
+		return 0, err
+	}
+
+	if len(result) > 0 {
+		id, ok := result[0]["id"].(float64)
+		if !ok {
+			err = fmt.Errorf("Error no id field in object: %+v", err)
+			return 0, err
+		}
+		return int(id), nil
+	}
+
+	return 0, fmt.Errorf("User %s not found", user.Login)
+}
+
+func (a *App) SetPermissionsHeader(userID int, settings OnCallPluginSettings, req *http.Request) error {
+	permissionsURL, err := url.JoinPath(settings.GrafanaURL, fmt.Sprintf("api/access-control/users/%d/permissions", userID))
```

Comment:
Is it working for you? Locally I needed to switch to 
`permissionsURL, err := url.JoinPath(settings.GrafanaURL, "api/access-control/user/permissions")` endpoint documented [here](https://grafana.com/docs/grafana/latest/developers/http_api/access_control/#list-your-permissions), otherwise I'm getting 404 (userId is correctly populated)

Reply:
Yes but I am running enterprise with RBAC locally. We can fallback to use the role when that is not enabled.